x/concentrated-liquidity/lp_test.go

```diff
@@ -551,6 +551,128 @@ func (s *KeeperTestSuite) TestIsInitialPosition() {
 	}
 }
 
+func (s *KeeperTestSuite) TestUpdatePosition() {
+	type updatePositionTest struct {
+		poolId                uint64
+		owner                 sdk.AccAddress
+		lowerTick             int64
+		upperTick             int64
+		liquidityDelta        sdk.Dec
+		amount0Expected       sdk.Int
+		amount1Expected       sdk.Int
+		expectedPoolLiquidity sdk.Dec
+		expectedError         bool
+	}
+
+	tests := map[string]updatePositionTest{
+		"update existing position with positive amount": {
+			poolId:                1,
+			owner:                 s.TestAccs[0],
+			lowerTick:             DefaultLowerTick,
+			upperTick:             DefaultUpperTick,
+			liquidityDelta:        DefaultLiquidityAmt,
+			amount0Expected:       DefaultAmt0Expected,
+			amount1Expected:       DefaultAmt1Expected,
+			expectedPoolLiquidity: DefaultLiquidityAmt.Add(DefaultLiquidityAmt),
+			expectedError:         false,
+		},
+		"error: update existing position with negative amount": {
```

Comment:
Why does this error?

If we remove liquidity that was added before by making `liquidityDelta` negative, it should pass

Reply:
We should have both error and success cases with negative liquidity but it is still unclear to me why this specific test case errors

Reply:
oh interesting seems that createPosition is not creating position 🤔 investigating further on

Reply:
oh wow the reason that caused this is actually pretty interesting, if we use s.TestAccs[0] in the field and use s.TestAccs[0] directly within the test cases, it would error since the suite setup sets the test accounts to random test accounts everytime

Reply:
Figured this out and added new tests for this